workers/worker_base.py

```diff
@@ -981,7 +1041,69 @@ def retrieve_tuple(self, key_values, tables):
             SELECT * FROM {} WHERE {}
             """.format(table_str, where_str))
         values = json.loads(pd.read_sql(retrieveTupleSQL, self.db, params={}).to_json(orient="records"))
-        return values  
+        return values
+
+    def update_gitlab_rate_limit(self, response, bad_credentials=False, temporarily_disable=False):
```

Comment:
In your opinion, do you think it is logical to combine update_gitlab_rate_limit and update_gh_rate_limit into a single update_rate_limit method with a `platform` kwarg? Is there enough similarity between the two to where that would make sense? I am asking because it may limit repetitive code.

Reply:
Compared to pagination, this the rate limit update function requires a lot more changes to it when comes to updating the gitlab rate limit. THe header keys received, the auth headers are completely different from that of github. This would incorporate a lot of unnecessary nesting at every line!

Reply:
Ok looks good then! Thank you for explaining